packages/filesystem/src/browser/file-tree/file-tree-widget.tsx

```diff
@@ -80,10 +81,16 @@ export class FileTreeWidget extends TreeWidget {
             onDragEnter: event => this.handleDragEnterEvent(node, event),
             onDragOver: event => this.handleDragOverEvent(node, event),
             onDragLeave: event => this.handleDragLeaveEvent(node, event),
-            onDrop: event => this.handleDropEvent(node, event)
+            onDrop: event => this.handleDropEvent(node, event),
+            title: this.getNodeTooltip(node)
         };
     }
 
+    protected getNodeTooltip(node: TreeNode): string | undefined {
+        const uri = UriSelection.getUri(node);
+        return uri && uri.path.toString();
```

Comment:
- Does this make sure it will be decoded?
 - Why the implicit `undefined` return? Can you make it explicit? AFAIK, that is the preferred way.

Reply:
- yes, tested. 

- sure, done :)